### PR TITLE
support encrypt column config data type

### PR DESCRIPTION
**Pull request by tuichenchuxin**
ref to #14987
- add encrypt rule config

### CODE REVIEW

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java [hunk 1]
```diff
@@ -18,22 +18,45 @@
...
+        this.logicColumn = logicColumn;
+        this.logicDataType = logicDataType;
+        this.cipherColumn = cipherColumn;
+        this.cipherDataType = cipherDataType;
+        this.assistedQueryColumn = assistedQueryColumn;
+        this.assistedQueryDataType = assistedQueryDataType;
+        this.plainColumn = plainColumn;
+        this.plainDataType = plainDataType;
+        this.encryptorName = encryptorName;
+    }
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java:61]: Why not use @RequiredArgsConstructor here?

Comment (tuichenchuxin) [reply shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java]: Yes, I'll replace the construct with @RequiredArgsConstructor.

Comment (tuichenchuxin) [reply shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java]: @terrymanu Hi, I have replace all args construct with @RequiredArgsConstructor. Please take a look when you are free.

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java [hunk 2]
```diff
@@ -23,17 +23,29 @@
 /**
  * Encrypt column rule configuration.
  */
-@RequiredArgsConstructor
 @Getter
+@RequiredArgsConstructor
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java:27]: Please change the sequence, we'd like use first line for Constructor, and second line for getter and setter

**Review by terrymanu — APPROVED** [review]

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java (hunk 2 above)
Comment (tuichenchuxin) [reply shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-api/src/main/java/org/apache/shardingsphere/encrypt/api/config/rule/EncryptColumnRuleConfiguration.java]: Thank you for reminding me. I have solved it.

**Pull request merged**